autoarray/plot/mat_wrap/wrap/wrap_base.py

```diff
@@ -0,0 +1,946 @@
+from autoconf import conf
+import matplotlib
+
+
+def set_backend():
+
+    backend = conf.get_matplotlib_backend()
+
+    if backend not in "default":
+        matplotlib.use(backend)
+
+    try:
+        hpc_mode = conf.instance["general"]["hpc"]["hpc_mode"]
+    except KeyError:
+        hpc_mode = False
+
+    if hpc_mode:
+        matplotlib.use("Agg")
+
+
+import matplotlib.pyplot as plt
+import matplotlib.colors as colors
+import matplotlib.cm as cm
+import numpy as np
+from os import path
+import os
+import typing
+
+from autoarray.structures import abstract_structure, arrays
+from autoarray import exc
+
+
+class Units:
+    def __init__(
+        self,
+        use_scaled: bool = None,
+        conversion_factor: float = None,
+        in_kpc: bool = None,
+    ):
+        """
+        This object controls the units of a plotted figure, and performs multiple tasks when making the plot:
+
+        1) Species the units of the plot (e.g. meters, kilometers) and contains a conversion factor which converts
+           the plotted data from its current units (e.g. meters) to the units plotted (e.g. kilometeters). Pixel units
+            can be used if `use_scaled=False`.
+
+        2) Uses the conversion above to manually override the yticks and xticks of the figure, so it appears in the
+           converted units.
+
+        3) Sets the ylabel and xlabel to include a string containing the units.
+
+        Parameters
+        ----------
+        use_scaled : bool
+            If True, plot the 2D data with y and x ticks corresponding to its scaled coordinates (its `pixel_scales`
+            attribute is used as the `conversion_factor`). If `False` plot them in pixel units.
+        conversion_factor : float
+            If plotting the labels in scaled units, this factor multiplies the values that are used for the labels.
+            This allows for additional unit conversions of the figure labels.
+        in_kpc : bool
+            If True, the scaled units are converted to kilo-parsecs via the input Comsology of the plot (this is only
+            relevent for the projects PyAutoGalaxy / PyAutoLens).
+        """
+
+        self.use_scaled = use_scaled
+        self.conversion_factor = conversion_factor
+        self.in_kpc = in_kpc
+
+        if use_scaled is not None:
+            self.use_scaled = use_scaled
+        else:
+            try:
+                self.use_scaled = conf.instance["visualize"]["general"]["general"][
+                    "use_scaled"
+                ]
+            except:
+                self.use_scaled = True
+
+        try:
+            self.in_kpc = (
+                in_kpc
+                if in_kpc is not None
+                else conf.instance["visualize"]["general"]["units"]["in_kpc"]
+            )
+        except:
+            self.in_kpc = None
+
+
+class AbstractMatWrap:
+    def __init__(self, kwargs: dict):

```

Comment:
Make kwargs **kwargs.

It will be of dict type anyway but this way it will be consistent with child classes